src/actions/heap/heap.ts

```diff
@@ -267,77 +307,112 @@ export class HeapAction extends Hub.Action {
     }, {} as LookerFieldMap)
   }
 
-  private resolveHeapField(propertyType: HeapPropertyType): HeapField {
+  private resolveHeapField(propertyType: HeapPropertyType, logTag: LogTag): HeapField {
     switch (propertyType) {
       case HeapPropertyTypes.Account:
         return HeapFields.AccountId
       case HeapPropertyTypes.User:
         return HeapFields.Identity
       default:
-        throw new Error(`Unsupported property type: ${propertyType}`)
+        const error = new Error(`Unsupported property type: ${propertyType}`)
+        winston.error(error.message, logTag)
+        throw error
     }
   }
 
-  private resolveApiEndpoint(propertyType: HeapPropertyType): string {
+  private resolveApiEndpoint(propertyType: HeapPropertyType, logTag: LogTag): string {
     switch (propertyType) {
       case HeapPropertyTypes.User:
         return HeapAction.ADD_USER_PROPERTIES_URL
       case HeapPropertyTypes.Account:
         return HeapAction.ADD_ACCOUNT_PROPERTIES_URL
       default:
-        throw new Error(`Unsupported property type: ${propertyType}`)
+        const error = new Error(`Unsupported property type: ${propertyType}`)
+        winston.error(error.message, logTag)
+        throw error
     }
   }
 
-  private extractRequiredField(row: Hub.JsonDetail.Row, field: Hub.Field): {key: string, value: any} {
+  private extractRequiredField(
+      row: Hub.JsonDetail.Row,
+      field: Hub.Field,
+    ): {
+      key: string,
+      value: string,
+      success: boolean,
+      error?: string,
+    } {
     // we have observed different behavior based on how the view is created
     // each of the following could potentially be used as keys of the field,
     // so we check each one until we find one that exists.
     const propertiesToCheck = [field.name, field.label_short, field.field_group_label]
     const fieldName = propertiesToCheck.find((f) => !!f && row.hasOwnProperty(f))
     if (fieldName) {
-      return {
+     const value = row[fieldName].value != null ? row[fieldName].value.toString() : ""
+     return {
         key: fieldName,
-        value: row[fieldName].value,
```

Comment:
value.toString is only called after we check if the returned value was falsy... rather than attaching the error to your returned value here, I think we should throw an error when the value is undefined and then handle that error in the calling function.